react/features/desktop-picker/components/DesktopPicker.tsx

```diff
@@ -325,14 +349,15 @@ class DesktopPicker extends PureComponent<IProps, IState> {
     _onTabSelected(id: string) {
         const { sources } = this.state;
 
-        this._selectedTabType = id;
-
         // When we change tabs also reset the screenShareAudio state so we don't
         // use the option from one tab when sharing from another.
         this.setState({
             screenShareAudio: false,
             selectedSource: this._getSelectedSource(sources),
-            selectedTab: id
+
+            // https://github.com/jitsi/jitsi-meet/pull/12994
+            // seleect type `window` or `screen` from id
+            selectedTab: id.split('-')[0]
```

Comment:
Fix bug introduced via https://github.com/jitsi/jitsi-meet/pull/12994 where id was changed and "-tab" was appended. Due to this the selected tab was not anymore working and empty object was returned here: https://github.com/dudumanbogdan/jitsi-meet/blob/05da37b56de391e3f67d5919a751a81a4e803d73/react/features/desktop-picker/components/DesktopPicker.tsx#L270

Reply:
Thank you for highlighting this issue and providing the fix, @dudumanbogdan.

Can you please resolve the conflicts that are introduced by the JS -> TS migration that recently happened, that would be awesome?